samples/cpp/digits.cpp

```diff
@@ -1,375 +1,185 @@
     {
-        deskewed_img = img.clone();
-        return;
+        net = dnn::readNetFromCaffe(modelTxt, modelBin);
     }
-
-    float skew = (float)(m.mu11 / m.mu02);
-    float M_vals[2][3] = {{1, skew, -0.5f * SZ * skew}, {0, 1, 0}};
-    Mat M(Size(3, 2), CV_32F);
-
-    for (int i = 0; i < M.rows; i++)
+    catch (cv::Exception &ee)
     {
-        for (int j = 0; j < M.cols; j++)
+        cerr << "Exception: " << ee.what() << endl;
+        if (net.empty())
         {
-            M.at<float>(i, j) = M_vals[i][j];
+            cout << "Can't load the network by using the flowing files:" << endl;
+            cout << "modelTxt: " << modelTxt << endl;
+            cout << "modelBin: " << modelBin << endl;
+            exit(-1);
         }
     }
 
-    warpAffine(img, deskewed_img, M, Size(SZ, SZ), WARP_INVERSE_MAP | INTER_LINEAR);
-}
+    static const string resultWinName = "LeNet Result";
+    static const string preWinName = "Preprocessing";
 
-static void mosaic(const int width, const vector<Mat>& images, Mat& grid)
-{
-    int mat_width = SZ * width;
-    int mat_height = SZ * (int)ceil((double)images.size() / width);
+    namedWindow(resultWinName, WINDOW_AUTOSIZE);
+    namedWindow(preWinName, WINDOW_AUTOSIZE);
 
-    if (!images.empty())
-    {
-        grid = Mat(Size(mat_width, mat_height), images[0].type());
+    Mat labels, img_color, stats, centroids;
+    Point positiosn;
 
-        for (size_t i = 0; i < images.size(); i++)
-        {
-            Mat location_on_grid = grid(Rect(SZ * ((int)i % width), SZ * ((int)i / width), SZ, SZ));
-            images[i].copyTo(location_on_grid);
-        }
-    }
-}
+    Rect getRectangle;
+    bool ifDrawingBox = false;
 
-static void evaluate_model(const vector<float>& predictions, const vector<Mat>& digits, const vector<int>& labels, Mat& mos)
-{
-    double err = 0;
+    int classId = 0;
+    double probability = 0;
 
-    for (size_t i = 0; i < predictions.size(); i++)
-    {
-        if ((int)predictions[i] != labels[i])
-        {
-            err++;
-        }
-    }
+    VideoCapture cap(0);
 
-    err /= predictions.size();
+    // Set camera resolution
+    cap.set(CAP_PROP_FRAME_WIDTH, vWidth);
+    cap.set(CAP_PROP_FRAME_HEIGHT, vHeight);
 
-    cout << format("error: %.2f %%", err * 100) << endl;
+    Rect basicRact = Rect(0, 0, 640, 480);
+    Mat rawImage;
 
-    int confusion[10][10] = {};
+    double fps = 0;
 
-    for (size_t i = 0; i < labels.size(); i++)
+    // Open a video file or an image file or a camera stream.
+    if (cap.isOpened())
     {
-        confusion[labels[i]][(int)predictions[i]]++;
-    }
+        TickMeter cvtm;
 
-    cout << "confusion matrix:" << endl;
-    for (int i = 0; i < 10; i++)
-    {
-        for (int j = 0; j < 10; j++)
+        while (true)
         {
-            cout << format("%2d ", confusion[i][j]);
-        }
-        cout << endl;
-    }
+            cvtm.reset();
+            cvtm.start();
+            cap >> rawImage;
```

Comment:
Please check for `rawImage.empty()` that means end of video sequence or camera disconnect.